upsolver-api/middleware/verifyTeamMembership.js

```diff
@@ -0,0 +1,32 @@
+import { sequelize } from "../database.js";
+import { Role } from "../models/role.js";
+import tryCatch from "../utils/tryCatch.js";
+import AppError from "../utils/AppError.js";
+
+const verifyTeamMembership = tryCatch(async (req, res, next) => {
+  const { username } = req.user;
+  const { teamId } = req.params;
+  const getUserRoleToTeam = `
+    SELECT 
+        R.name 
+    FROM
+        teams T
+        INNER JOIN roles R
+            ON T.id = R.teamId
+        INNER JOIN users U
+            ON R.userId = U.id
+    WHERE
+        U.username = '${username}'
+        AND T.id = ${teamId};`;
```

Comment:
quick question: is there a way we can do it with sequelize's query builder? or maybe the joins are easier to express in pure sql?

the query is ok but usually we'd want a query builder - the problem with raw queries like this is that they are subject to security issues like sql injection